### DISCUSSION TITLE
Creating single platform on Generator

### DISCUSSION THREAD

**Discussion by ilsubyeega** [Minestom/Minestom#1240] [Help]
When creating an instance, i wanted to create single platform, however there are two ways to do that i figured out.
1. Creating instance and then preload the chunks, and place blocks manually by batch.
2. Using Generator by like this code below.

It seems that `unit.fork` run every time when generate is invoked, which is best solution to create single platform?
```kotlin
class PlatformWorldGenerator(val size: Int = 10, val yPos: Int = 0, val block: Block = Block.STONE) : Generator {
    override fun generate(unit: GenerationUnit) {
        unit.fork {
            for (xPos in -size..size)
                for (zPos in -size..size) {
                    it.setBlock(xPos, yPos, zPos, block)
                }
        }
    }
}
```

**Comment by Flopgop** [comment]
Try checking if the unit.absoluteStart.blockX and blockY is 0 before generating, therefore it will only generate if the chunk being generated is at 0,0
